hash_practice/exercises.py

```diff
@@ -1,29 +1,164 @@
 
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
-        Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Each subarray will have strings which are anagrams of each other.
+        Time Complexity: O(n^2) -- or is it O(n) bc of the "continue"s?
+        Space Complexity: O(n)
     """
-    pass
+    result = []
+
+    for i in range(len(strings)):
+        if strings[i] == None:
+            continue
+
+        anagrams = [strings[i]]
+        strings[i] = None
+
+        for j in range(len(strings)):
+            if strings[j] == None:
+                continue
+
+            if __is_anagram(anagrams[0], strings[j]):
+                anagrams.append(strings[j])
+                strings[j] = None
+
+        result.append(anagrams)
+
+    return result
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n x m), where n is length of original list and m is number of repeats for each repeated value
+        Space Complexity: O(n)

```

Comment:
👍 Just note that you are sorting, what does that do to the time/space complexity?